### PR TITLE
Test calling finished if one of the streams closes prematurely

### PR DESCRIPTION
**Pull request by Vunovati**
Cover the case when one of the streams gets closed prematurely after Fastify had received it.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Vunovati** [comment]
I was still unable to cover the last remaining two lines (282,283) with tests. Here are my premises and conclusion why I think that is.

Two lines in `waitForFiles` are executed when:
`!err && !completed && callDoneOnNextEos && (count === files)`

According to my understanding (and observations), busboy `finish` event will only be fired after the file streams it contains have been closed. EOS’s callback, `waitForFiles`, will be invoked after those file streams have ended. 
That means that busboy finish event will always be fired after all the `waitForFiles` has been called.

Since busboy finish callback is the only one that can set the `callDoneOnNextEos` to true, the aforementioned order of events would mean that `callDoneOnNextEos` will never be true when evaluated in waitForFiles and testing lines 282,283 would not be possible:
```
      if (callDoneOnNextEos && count === files) {
        completed = true
        done()
      }
```
If I am correct, then these lines would be unreachable for all cases.

**Comment by mcollina** [comment]
I understand! Go ahead and simplify all that logic removing both `count` and `callDoneOnNextEos`.

File: test/legacy/multipart.test.js [hunk 1]
```diff
@@ -139,6 +139,78 @@ test('should call finished when both files are pumped', { skip: process.platform
   })
 })
 
+test('should call finished if one of the streams closes prematurely', { skip: process.platform === 'win32' }, function (t) {
+  t.plan(5)
+
+  const fastify = Fastify({
+    logger: {
+      level: 'debug'
+    }
```
Comment (simoneb) [inline test/legacy/multipart.test.js:148]: remove logging as we don't want the tests to output anything

**Review by simoneb — APPROVED** [review]
One comment, LGTM

File: test/legacy/multipart.test.js (hunk 1 above)
Comment (Vunovati) [reply test/legacy/multipart.test.js:148]: Done

**Review by simoneb — APPROVED** [review]

**Pull request merged**